RecoLocalCalo/HGCalRecAlgos/src/HGCalImagingAlgo.cc

```diff
@@ -197,7 +197,7 @@ math::XYZPoint HGCalImagingAlgo::calculatePosition(std::vector<KDNode> &v){
 			 z/total_weight );
     }
   }
-  else {
+  else if (v_size > 0) {
```

Comment:
perhaps better would be to set maxEnergyIndex to -1 or other value that means uninitialized and then check it here.